documents/development/README.md

```diff
@@ -0,0 +1,539 @@
+- There should be `celix::` prefixed aliases for the library.
+- C++ Libraries should support C++14.
+  - Exception are `celix::Promises` and `celix::PushStreams` which requires C++17.
+- C++ Libraries should prefer to be header-only.
+- C++ support for `celix::framework` and `celix::utils` must be header-only.
+- Header-only C++ libraries do not need an export header and do not need to configure symbol visibility.
+- C++ shared libraries (lib with C++ sources), should configure an output name with a `celix_` prefix.
+- C++ shared libraries (lib with C++ sources), should use an export header and configure symbol visibility.
+  - See the C Libraries section for more information.
+
+### C++ Services
+
+- Use `CamelCase` (starting with a capital) for service names.
+- Add a 'I' prefix to the service interface name.
+- Place service classes in a `celix::` namespace or sub `celix::` namespace.
+- Add a `static constexpr const char * const NAME` to the service class, for the service name.
+- Add a `static constexpr const char * const VERSION` to the service class, for the service version.
+
+### C++ Bundles
+
+- Use `CamelCase` for C++ bundle target names.
+- Do _not_ use a `Celix` prefix for C++ bundle target names.
+- Use `celix::` prefixed aliases for C++ bundle targets.
+- Use `CamelCase` for C++ bundle symbolic names.
+- Configure at least SYMBOLIC_NAME, NAME, FILENAME, VERSION and GROUP for C++ bundle targets.
+- Use `Apache_Celix_` prefix for C++ bundle symbolic names.
+- Use `Apache Celix ` prefix for C++ bundle names.
+- Use a `celix_` prefix for C++ bundle filenames.
+- Use a group name starting with `celix/` for C++ bundle groups.
+
+Examples:
+```cmake
+add_celix_bundle(MyBundle
+    SOURCES src/MyBundle.cc
+    SYMBOLIC_NAME "Apache_Celix_MyBundle"
+    NAME "Apache Celix My Bundle"
+    FILENAME "celix_MyBundle"
+    VERSION "1.0.0"
+    GROUP "celix/MyBundleGroup"
+)
+add_library(celix::MyBundle ALIAS MyBundle)
+```
+
+### Unit Tests Naming
+
+- The test fixture should have a`TestSuite` postfix.
+- The source file should be named after the test fixture name and use a `.cc` extension.
+- Testcase names should use `CamelCase` (starting with a capital) and have a `Test` postfix.
+- When using error injection (one of the `error_injector` libraries) a separate test suite should be used.
+  - A `ErrorInjectionTestSuite` postfix should be used for the test fixture.
+  - The error injection setup should be reset on the `TearDown` function or destructor of the test fixture.
+
+## Comments and Documentation
+
+- Use Doxygen for code documentation.
+- Write comments that explain the purpose of the code, focusing on the "why" rather than the "how".
+- Apply doxygen documentation to all public API's.
+- Use the javadoc style for doxygen documentation.
+- Use `@` instead of `\` for doxygen commands.
+- Start with a `@brief` command and a short description.
+- For `@param` commands also provide in, out, or in/out information.
```

Comment:
Good point, maybe some example of good doxygen usage should be included. 

For `man 2 write` return documentation you mean to  full paragraph:

> On success, the number of bytes written is returned.  On error, -1 is returned, and errno is set to indicate the error.

 > Note that a successful write() may transfer fewer than count bytes.  Such partial writes can occur for various reasons; for example, because there was insufficient space on the disk device to write all of the requested bytes, or because a blocked write() to a socket, pipe, or similar was interrupted by a signal handler after it had transferred some, but  before
       it had transferred all of the requested bytes.  In the event of a partial write, the caller can make another write() call to transfer the remaining bytes.  The subsequent call will either transfer further bytes or may result in an error (e.g., if the disk is now full).

> If count is zero and fd refers to a regular file, then write() may return a failure status if one of the errors below is detected.  If no errors are detected, or error detection is not performed, 0 is returned without causing any other effect.  If count is zero and fd refers to a file other than a regular file, the results are not specified.

or just the first sentence?

Reply:
And its `ERRORS` section.